cores/esp32/HardwareSerial.h

```diff
@@ -98,16 +98,13 @@ class HardwareSerial: public Stream
     uint32_t baudRate();
     operator bool() const;
 
-    size_t setRxBufferSize(size_t);
     void setDebugOutput(bool);
     
     void setRxInvert(bool);
 
 protected:
     int _uart_nr;
     uart_t* _uart;
-    uint8_t _tx_pin;
-    uint8_t _rx_pin;
```

Comment:
we actually had those for a reason. One might call in `setup()` `Serial.begin()` with custom RX/TX and then some library to call just `Serial.begin()`. In that case the pins will be reset to defaults. You should also think of adding `HadwareSerial::setPins(rx, tx)` in order to be able to control better which pins would be default.

Reply:
The reason for having _tx_pin and _rx_pin was to be able to call uartEnd() in HAL, which pins should be unattched when HardwareSerial::end() was called.  
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/HardwareSerial.cpp#L122
Given that it uses IDF, that data i  not necessary anymore.

Serial1.begin(), for instance, with no parameters, would never compile because there is no default baudrate in
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/HardwareSerial.h#L58
As well as this is not the default Arduino behavior, as described in
https://www.arduino.cc/reference/en/language/functions/communication/serial/begin/

When using, Serial.begin(115200), it will use default RX/TX pins for that specific SoC and UART Number.

To control RX, TX pins, it should be done, for instance, by:
Serial1.begin(115200, SERIAL_8N1, RxPin, TxPin);  // can add more parameters if necessary

HardwareSerial:setPins(rx, tx) can be implemented using IDF.

Summary: 
Implement HardwareSerial:setPins(rx, tx)

Reply:
Of course begin will be called with the baudrate... did not think would be necessary to add that in order to explain the problem.

Reply:
In that case it works fine, because not passing rxPin and txPin will reset the UART port to the deafult pins. There is no change to be done neither it needs to keep _tx_pin and _rx_pin.

I have implemented HardwareSerial:setPins(rx, tx). I'll test it and then commit it.

Reply:
HardwareSerial.setPins(rx, tx)  has been implemented.
Done. Commited.